.github/workflows/deploy-infra.yml

```diff
@@ -0,0 +1,51 @@
+name: DeployInfra
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'infrastructure/parameter/**'
+jobs:
+  deploy:
+    name: DeployInfra
+    runs-on: ubuntu-latest
+    environment: di-ipv-cri-address-build
+    timeout-minutes: 15
+    env:
+      AWS_REGION: eu-west-2
+      ENVIRONMENT: build
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup SAM
+        uses: aws-actions/setup-sam@v1
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.PARAM_GH_ACTIONS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: SAM Validate
```

Comment:
Do we need to do another sam validate here?

Reply:
We don't have another at present, it is disabled: The PR validation of the SAM infrastructure cannot run because it requires to assume the AWS role, however it is not running on the main branch so correctly is denied access to the environment secrets. We need an offline validator for this, I have disabled the validation workflow for now and added the validation into the deployment here for now.